tests-fuzz/targets/migration/fuzz_migrate_metric_regions.rs

```diff
@@ -0,0 +1,495 @@
+                "expected rows affected: {}, actual: {}",
+                rows,
+                result.rows_affected(),
+            )
+        }
+    );
+
+    Ok(insert_expr)
+}
+
+#[derive(Debug, Clone, Copy)]
+struct Migration {
+    from_peer: u64,
+    to_peer: u64,
+    region_id: RegionId,
+}
+
+async fn execute_migration(ctx: FuzzContext, input: FuzzInput) -> Result<()> {
+    let mut rng = ChaCha20Rng::seed_from_u64(input.seed);
+    // Creates a physical table.
+    let physical_table_ctx = create_physical_table(&ctx, &mut rng).await?;
+    let mut tables = HashMap::with_capacity(input.tables);
+
+    let table_num = (input.tables / 3).max(1);
+
+    for _ in 0..table_num {
+        let translator = CreateTableExprTranslator;
+        let create_logical_table_expr =
+            generate_create_logical_table_expr(physical_table_ctx.clone(), &mut rng).unwrap();
+        if tables.contains_key(&create_logical_table_expr.table_name) {
+            // Ignores same name logical table.
+            continue;
+        }
+        let sql = translator.translate(&create_logical_table_expr)?;
+        let result = sqlx::query(&sql)
+            .execute(&ctx.greptime)
+            .await
+            .context(error::ExecuteQuerySnafu { sql: &sql })?;
+        info!("Create logical table: {sql}, result: {result:?}");
+        let logical_table_ctx = Arc::new(TableContext::from(&create_logical_table_expr));
+
+        let insert_expr =
+            insert_values(input.rows, &ctx, &mut rng, logical_table_ctx.clone()).await?;
+        if rng.gen_bool(0.1) {
+            flush_memtable(&ctx.greptime, &physical_table_ctx.name).await?;
+        }
+        if rng.gen_bool(0.1) {
+            compact_table(&ctx.greptime, &physical_table_ctx.name).await?;
+        }
+
+        tables.insert(
+            logical_table_ctx.name.clone(),
+            (logical_table_ctx.clone(), insert_expr),
+        );
+    }
+
+    // Fetches region distribution
+    let partitions = fetch_partitions(&ctx.greptime, physical_table_ctx.name.clone()).await?;
+    let num_partitions = partitions.len();
+    let region_distribution = region_distribution(partitions);
+    info!("Region distribution: {region_distribution:?}");
+    let datanodes = fetch_nodes(&ctx.greptime)
+        .await?
+        .into_iter()
+        .flat_map(|node| {
+            if node.peer_type == PEER_TYPE_DATANODE {
+                Some(node)
+            } else {
+                None
+            }
+        })
+        .collect::<Vec<_>>();
+    info!("List datanodes: {:?}", datanodes);
+
+    // Generates region migration task.
+    let mut migrations = Vec::with_capacity(num_partitions);
+    let mut new_distribution: BTreeMap<u64, HashSet<_>> = BTreeMap::new();
+    for (datanode_id, regions) in region_distribution {
+        let step = rng.gen_range(1..datanodes.len());
+        for region in regions {
+            let to_peer = (datanode_id + step as u64) % datanodes.len() as u64;
```

Comment:
Is it possible that the `to_peer` is the same as the current datanode id?

Reply:
Nope